doc/source/configuration/firewall.rst

```diff
@@ -0,0 +1,350 @@
+
+Custom rules
+------------
+
+Custom firewalld rules can be added for any of the following groups using their
+corresponding variables:
+
+* Controllers - ``stackhpc_controller_firewalld_rules_extra``
+* Compute - ``stackhpc_compute_firewalld_rules_extra``
+* Storage - ``stackhpc_storage_firewalld_rules_extra``
+* Monitoring - ``stackhpc_monitoring_firewalld_rules_extra``
+* Wazuh Manager Infrastructure VM - ``stackhpc_wazuh_manager_infra_vm_firewalld_rules_extra``
+* Ansible Control host Infrastructure VM - ``stackhpc_ansible_control_infra_vm_firewalld_rules_extra``
+* Seed - ``stackhpc_seed_firewalld_rules_extra``
+* Seed Hypervisor - ``stackhpc_seed_hypervisor_firewalld_rules_extra``
+
+Each variable is a list of firewall rules to apply. Each item is a dict
+containing arguments to pass to the firewalld module.
+
+The example below would enable SSH on the ``provision_oc`` network, and disable
+UDP port 1000 on the ``admin_oc`` network for the Wazuh manager Infrastructure
+VM:
+
+.. code-block:: yaml
+   :caption: ``etc/kayobe/inventory/group_vars/wazuh_manager/firewall``
+
+   stackhpc_wazuh_manager_infra_vm_firewalld_rules_extra:
+     -  service: ssh
+        network: "{{ provision_oc_net_name }}"
+        zone: "{{ provision_oc_net_name | net_zone }}"
+        state: enabled
+     -  port: 1000/udp
+        network: "{{ admin_oc_net_name }}"
+        zone: "{{ admin_oc_net_name | net_zone }}"
+        state: disabled
+
+Beware that if any rules are found that directly conflict (a service or port is
+both enabled and disabled) the configuration will fail. There is currently no
+way to override rules in the standard configuration, other than to find the
+rule and delete it manually. If you find a standard rule that does not work for
+your deployment, please consider merging your changes back in to upstream SKC.
+
+Applying changes
+----------------
+
+The ``kayobe configuration dump`` command can be used to view all the rules
+that will be applied to a host.
+
+.. code-block:: bash
+
+   kayobe configuration dump --var-name stackhpc_firewalld_rules --limit <host>
+
+If the command above prints a template, rather than a clean list of rules, the
+configuration is invalid. The kayobe configuration dump command can be used on
+other variables such as ``stackhpc_firewalld_rules_unverified`` or
+``stackhpc_*_firewalld_rules`` to debug the configuration. See the `How it
+works`_ section for more details.
+
+Ensure Kolla Ansible opens up ports in firewalld for services on the public
+API network:
+
+.. code-block:: yaml
+   :caption: ``etc/kayobe/kolla/globals.yml``
+
+   enable_external_api_firewalld: true
+   external_api_firewalld_zone: "{{ public_net_name | net_zone }}"
+
+Ensure every network in ``networks.yml`` has a zone defined. The standard
+configuration is to set the internal network zone to ``trusted`` and every
+other zone to the name of the network. See
+``etc/kayobe/environments/ci-multinode/networks.yml`` for a practical example.
```

Comment:
```suggestion
Prerequisites
----------------

Ensure Kolla Ansible opens up ports in firewalld for services on the public
API network:

.. code-block:: yaml
   :caption: ``etc/kayobe/kolla/globals.yml``

   enable_external_api_firewalld: true
   external_api_firewalld_zone: "{{ public_net_name | net_zone }}"

Ensure every network in ``networks.yml`` has a zone defined. The standard
configuration is to set the internal network zone to ``trusted`` and every
other zone to the name of the network. See
``etc/kayobe/environments/ci-multinode/networks.yml`` for a practical example.

Applying changes
----------------

The ``kayobe configuration dump`` command can be used to view all the rules
that will be applied to a host.

.. code-block:: bash

   kayobe configuration dump --var-name stackhpc_firewalld_rules --limit <host>

If the command above prints a template, rather than a clean list of rules, the
configuration is invalid. The kayobe configuration dump command can be used on
other variables such as ``stackhpc_firewalld_rules_unverified`` or
``stackhpc_*_firewalld_rules`` to debug the configuration. See the `How it
works`_ section for more details.
```

Reply:
Good point. I've gone one step further and split out the KA config and config validation